cpp/src/arrow/util/tdigest.cc

```diff
@@ -21,6 +21,7 @@
 #include <cmath>
 #include <iostream>
 #include <queue>
+#include <tuple>
```

Comment:
Thanks for fixing it.
It's strange why CI doesn't find this error when I uploaded the original patch.

Reply:
Maybe, `-DCMAKE_UNITY_BUILD=ON` hides this problem.